### PR TITLE
Add X-Forwarded-For header name

### PR DESCRIPTION
**Pull request by DenisBalan**
tldr; add header name `X-Forwarded-For` as constant in `HeaderNames` static class.

more info https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For

### CODE REVIEW

**Comment by davidfowl** [comment]
Why? We have middleware that already translates this into the appropriate values.

**Comment by DenisBalan** [comment]
> 
> 
> Why? We have middleware that already translates this into the appropriate values.

This is useful when creating own http calls to other services, to not have magic strings, but reference to one source of truth ie `HeaderNames`

File: src/Http/Headers/src/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -370,4 +370,5 @@ static readonly Microsoft.Net.Http.Headers.HeaderNames.WWWAuthenticate -> string
 static readonly Microsoft.Net.Http.Headers.HeaderNames.Warning -> string!
 static readonly Microsoft.Net.Http.Headers.HeaderNames.WebSocketSubProtocols -> string!
 static readonly Microsoft.Net.Http.Headers.HeaderNames.XFrameOptions -> string!
+static readonly Microsoft.Net.Http.Headers.HeaderNames.XForwardedFor -> string!
```
Comment (benaadams) [inline src/Http/Headers/src/PublicAPI.Shipped.txt:373]: Should be in `PublicAPI.Unshipped.txt` not Shipped

**Comment by Tratcher** [comment]
@dotnet/aspnet-api-review, I flagged this PR because I want to discuss and document what our policy is for adding constants to a list like this.

**Comment by Tratcher** [comment]
@DenisBalan we reviewed this and decided to add a written policy for what should and should not be added to this list. See https://github.com/dotnet/aspnetcore/pull/32394/files. Based on that policy I'm going to close this PR. Thanks for bringing this to our attention.

**Pull request closed without merging**